packages/bloc/test/bloc_on_test.dart

```diff
@@ -62,6 +62,16 @@ void main() {
       expect(() => DuplicateHandlerBloc(), expected);
     });
 
+    test('throws StateError when handler is missing', () {
+      const expectedMessage =
+          '''add(TestEventA) was called without a registered event handler.\n'''
+          '''Make sure to register a handler via on<TestEventA>((event, emit) {...})''';
+      final expected = throwsA(
+        isA<StateError>().having((e) => e.message, 'message', expectedMessage),
+      );
+      expect(() => MissingHandlerBloc()..add(TestEventA()), expected);
```

Comment:
Can you explain the cascade notation here? What does the second dot notation return?

Reply:
Check out https://dart.dev/guides/language/language-tour#cascade-notation.

Reply:
Good catch, the cascade operator is unnecessary in this case 👍